js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -614,6 +614,25 @@ ICInterpretOps(BaselineFrame* frame, VMFrameManager& frameMgr, State& state,
     DISPATCH_CACHEOP();
   }
 
+  CACHEOP_CASE(GuardToInt32ModUint32) {
+    ValOperandId inputId = icregs.cacheIRReader.valOperandId();
+    Int32OperandId resultId = icregs.cacheIRReader.int32OperandId();
+    BOUNDSCHECK(resultId);
+    Value input = Value::fromRawBits(icregs.icVals[inputId.id()]);
+    if (input.isInt32()) {
+      icregs.icVals[resultId.id()] = Int32Value(input.toInt32()).asRawBits();
+      DISPATCH_CACHEOP();
+    } else if (input.isDouble()) {
+      double doubleVal = input.toDouble();
+      if (doubleVal >= double(INT64_MIN) && doubleVal <= double(INT64_MAX)) {
+        icregs.icVals[resultId.id()] =
+            Int32Value(int64_t(doubleVal)).asRawBits();
+        DISPATCH_CACHEOP();
+      }
+    }
+    return ICInterpretOpResult::NextIC;
```

Comment:
just a minor style thing, but I'd prefer to invert the fallthrough-ness here, if you don't mind: make the `if` and `else if` cases fallthrough to a single `DISPATCH_CACHEOP()`, and add an `else { return NextIC; }`. Somehow that reads more clearly to me -- happy/successful path runs to the end, others depart early.

Reply:
Ah, sorry for not recognizing that and for not following my own coding tastes! :-) If you'd like, feel free to invert `GuardToInt32Index`'s control flow as well.